packages/cli-platform-ios/src/commands/runIOS/index.ts

```diff
@@ -230,7 +234,10 @@ async function runOnSimulator(
           findMatchingSimulator(simulators, {simulator: fallback})
         );
       },
-      findMatchingSimulator(simulators, args),
+      findMatchingSimulator(simulators, {
+        ...args,
+        simulator: args.simulator ?? defaultSimulator,
```

Comment:
Something along the lines
```suggestion
        simulator: args.simulator ?? fallbackSimulators[0],
```